### PR TITLE
feat: add a skip option for useSubscription

### PR DESCRIPTION
**Pull request by n1ru4l**
Sometimes it is useful to skip a subscription as the variables for setting up the subscription might not be available yet.

Closes https://github.com/relay-tools/relay-hooks/issues/131

Test will follow soon.

### CODE REVIEW

File: src/useSubscription.ts [hunk 1]
```diff
@@ -2,13 +2,22 @@ import { useEffect } from 'react';
...
+type SubscriptionConfig = {
+  skip?: boolean;
+}
+
 export function useSubscription<TSubscriptionPayload extends OperationType = OperationType>(
     config: GraphQLSubscriptionConfig<TSubscriptionPayload>,
+    opts?: SubscriptionConfig
 ): void {
     const environment = useRelayEnvironment();
+    const skip = opts?.skip ?? false
```
Comment (morrys) [inline src/useSubscription.ts:14]: Here I would prefer to declare the constant like this: **const skip = opts && opts.skip;** 
As I find it simple and it reduces the bundle size :)

File: src/useSubscription.ts [hunk 2]
```diff
@@ -2,13 +2,22 @@ import { useEffect } from 'react';
...
 
     useEffect(() => {
+        if (skip) {
+          return
+        }
```
Comment (morrys) [inline src/useSubscription.ts:19]: @n1ru4l try using `yarn format` in order to resolve eslint check

**Review by morrys — CHANGES_REQUESTED** [review]

**Comment by n1ru4l** [comment]
@morrys I was also wondering how we could type the `useSubscription` arguments in a way whether the `variables` parameter is optional if `skip` is true.

**Comment by morrys** [comment]
> @morrys I was also wondering how we could type the useSubscription arguments in a way whether the variables parameter is optional if skip is true.

@n1ru4l, you can try to change this https://github.com/relay-tools/relay-hooks/blob/b35e16b5132500dc75d92f02a012cc32c811c3a2/src/useSubscription.ts#L1-L12

with:
```ts
import { useEffect } from 'react';
import {
    GraphQLSubscriptionConfig,
    requestSubscription,
    OperationType,
    GraphQLTaggedNode,
} from 'relay-runtime';
import { useRelayEnvironment } from './useRelayEnvironment';

export type SubscriptionConfig = {
    skip: true;
};

export interface SkipGraphQLSubscriptionConfig<TSubscription extends OperationType>
    extends Omit<GraphQLSubscriptionConfig<TSubscription>, 'variables' | 'subscription'> {
    subscription?: GraphQLTaggedNode;
    variables?: TSubscription['variables'];
}

export function useSubscription<TSubscriptionPayload extends OperationType = OperationType>(
    config: GraphQLSubscriptionConfig<TSubscriptionPayload>,
): void;
export function useSubscription<TSubscriptionPayload extends OperationType = OperationType>(
    config: SkipGraphQLSubscriptionConfig<TSubscriptionPayload>,
    opts: SubscriptionConfig,
): void;
export function useSubscription<TSubscriptionPayload extends OperationType = OperationType>(
    config: GraphQLSubscriptionConfig<TSubscriptionPayload>,
    opts?: SubscriptionConfig,
): void {
```

let me know

**Comment by morrys** [comment]
Now I think it's okay without causing any breaking changes :)

Can I ask you to move [all type declarations](https://github.com/relay-tools/relay-hooks/blob/f32214770f7d06ac8861cc34f2f792f7e09d5f43/src/useSubscription.ts#L5-L17) into [RelayHooksTypes.ts](https://github.com/relay-tools/relay-hooks/blob/f32214770f7d06ac8861cc34f2f792f7e09d5f43/src/RelayHooksTypes.ts)?

**Comment by morrys** [comment]
now it seems perfect 👍 

now I'm not on the PC, Monday I accept it :) 

thanks for your work

**Pull request merged**